controllers/user/password_reset/impl/resetUserPassword.js

```diff
@@ -25,21 +28,35 @@ module.exports = (req, res) => {
             })
             //console.log(resetToken)
         
-            ResetPassword.create(resetToken, (err) =>{ //create new reset password
+            ResetPassword.create(resetToken, async (err) =>{ //create new reset password
                 if(err){
                     console.log("Error Creating Reset Password Schema")
                 }else{
                     // create reusable transporter object using the default SMTP transport
-                    let transporter = nodemailer.createTransport({
-                        host: 'smtp.gmail.com',
-                        port: 465,
-                        secure: true, // true for 465, false for other ports
+                    const oauth2Client = new OAuth2(
```

Comment:
same here, seperate function,  will look cleaner

Reply:
again, can't tell where you want to break up code, each piece of code is part of the previous callback, separating them into functions will only reduce the readability since following functions will have to be called within previous functions